app/src/main/java/fr/free/nrw/commons/upload/UploadService.java

```diff
@@ -297,7 +298,7 @@ private void onSuccessfulUpload(Contribution contribution, UploadResult uploadRe
     private void saveCompletedContribution(Contribution contribution, UploadResult uploadResult) {
         compositeDisposable.add(mediaClient.getMedia("File:" + uploadResult.getFilename())
         .map(media -> new Contribution(media, Contribution.STATE_COMPLETED))
-        .subscribe(newContribution -> contributionDao.deleteAndSaveContribution(contribution, newContribution)));
+        .subscribe(newContribution -> contributionDao.saveAndDelete(contribution, newContribution)));
```

Comment:
this won't work, `saveAndDelete` is a completable.

`subscribe` is a terminal call, it receives the values from the particular stream it subscribes to but it cannot continue the stream.

you probably want to use `.flatMapCompletable`

Reply:
Just updated the PR.